doc/tutorial/hello_world.dox

```diff
@@ -0,0 +1,165 @@
+
+/**
+ * @page hello-world Hello world
+ * @tableofcontents
+ *
+ * @section prerequisites Prerequisites
+ * @snippet tutorial.c Prerequisites
+ *
+ * The first step in creating a DPS application is to include the
+ * necessary header files.
+ *
+ * @section creating-a-node Creating a node
+ * @snippet tutorial.c Creating a node
+ *
+ * Each entity in DPS is represented by a @c DPS_Node.  The node may
+ * be a publisher, subscriber, both, or neither.  For this example,
+ * we're going to be creating publisher and subscriber nodes.
+ *
+ * Creating a node requires three parameters: the topic separators, a
+ * key store, and a key identifier.  For now we're only concerned with
+ * the separators.  Key stores and identifiers are covered later when
+ * discussing how to secure communications.
+ *
+ * The separators parameter is a string containing the characters used
+ * as topic level separators.  A topic is composed of multiple levels
+ * with separators between them.  Providing @c /. as the separators
+ * parameter value means all of the following topics are equivalent:
+ * @code
+ * a/b/c/d
+ * a.b.c.d
+ * a/b.c/d
+ * @endcode
+ *
+ * @see DPS_SetNodeData(), DPS_GetNodeData()
+ *
+ * @section starting-a-node Starting a node
+ * @snippet tutorial.c Starting a node
+ *
+ * Once created, a node must be started.  Starting a node enables it
+ * to begin sending and receiving DPS messages in the network.
+ *
+ * For this example, we are going to be sending and receiving
+ * multicast publications so we enable both and let DPS assign the
+ * listening port.
+ *
+ * @see DPS_MCAST_PUB_DISABLED, DPS_GetPortNumber()
+ *
+ * @section publishing Publishing
+ *
+ * @subsection creating-a-publication Creating a publication
+ * @snippet tutorial.c Creating a publication
+ *
+ * Each publication in DPS is represented by a @c DPS_Publication.
+ * Each publication has a set of topics, a UUID, and a sequence
+ * number.  In this example we are creating a publication with one
+ * topic, @c a/b/c/d.  The UUID is assigned by DPS and the sequence
+ * number will be incremented each time we publish.
+ *
+ * The @c noWildCard parameter is used by the publisher to control
+ * whether a subscription is required to match the publication's
+ * topics exactly or can use wildcards to match the topics.  If we set
+ * @c noWildCard to @c DPS_TRUE then only a subscription to @c a/b/c/d
+ * will receive this publication.  Since we set @c noWildCard to @c
+ * DPS_FALSE here, subscriptions to @c a/#, @c a/+/+/d, or similar
+ * variations will receive this publication.
```

Comment:
are the wildcard variations covered elsewhere?

Reply:
No, I'll add a brief topics section and describe the DPS wildcards.

Reply:
I'll add the why